homeassistant/components/raspihats/switch.py

```diff
@@ -95,6 +109,11 @@ def __init__(self, board, address, channel, name, invert_logic, initial_state):
                 state = initial_state
             self.I2C_HATS_MANAGER.write_dq(self._address, self._channel, state)
 
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+        if TYPE_CHECKING:
+            assert self.I2C_HATS_MANAGER
+
         def online_callback():
             """Call fired when board is online."""
             self.schedule_update_ha_state()
```

Comment:
We should probably move the `read_dq` call here from `is_on` as that call waits for a lock and most likely does I/O. Then we should call this `online_callback` in the service calls if we want to keep the call flow as it is now. So I'd move `online_callback` to an instance method.